### ISSUE TITLE
Timeouts on the readiness probe and liveness are switched

### ISSUE THREAD

**Issue by sjavurek** [syndesisio/syndesis#2385]
## This is a...

<pre><code>
[ ] Feature request
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ x] Bug report  
[ ] Documentation issue or request
</code></pre>

## The problem
The timeouts for the probes are backwards. The readiness probe is every 1 second while the liveness probe is 5 minutes. 

## Expected behavior

Reverse the settings so they are more realistc

## Screenshot
![readiness](https://user-images.githubusercontent.com/650863/38755907-87d78580-3f35-11e8-81d8-90bc701688a6.jpg)

**Comment by dsimansk** [comment]
@sjavurek the 10s and 300s part is the initial delay when to start probing after deployment is triggered. That should be all good, afaik.
@rhuss any comments on this?

**Comment by rhuss** [comment]
@sjavurek @dsimansk right, the delay is really when the check should be called *first*.

Remember, liveness probes when unsuccessful for five timed in a row (default) they will kill the pod. As we deal here with Java apps with a potential long startup time, 5 mins is a fair delay for starting to check for this liveness (actually its a good enough default for the various environments (oso, minishift, ocp, ...) we support, obviously this could be optimised for a specific cluster environment).

Readiness is different, as it "just" indicates when the the pod can be used in a service. (it doesn't kill any pods). So, a readiness check can kick in much earlier than the liveness probe to be ready more quickly if the pod starts faster than 5 mins.

I.e. when we would have a liveness check initial delay of 10s, with the default failure threshold of 3 and a period of 20s then, if the spring-boot pod doesn't comes up after 10s + 3 * 20s = 70s then it will be **killed**. And we know, spring-boot services easily can take longer to startup (much longer even).

The '1s' is just a timeout how long we want to wait for a network based check to succeed, so it's not a polling interval.

imo the values are chosen careful with the experience we gained over the time. Happy to improve here of course, though.

For the reference the concrete values for Java (spring-boot) based services are:

* Liveness:
  - initial delay:  5mins
  - check period: 20s
  - failureThreshold: 5 (i.e. after how many failed checks the pod should be restarted)
* Readiness:
  - initial delay: 10s (so that the service quickly gets accessible when pod is up)

All other values are the default. And for other services, we have other handcrafted values.

'hope this clarifies a bit the selection we made, but happy to discuss those, of course.

**Comment by gashcrumb** [comment]
This is a great write up, thanks for that.  I think the main confusion actually came from how the openshift console displays the probe values, it's not terribly obvious.

We were wondering though if we should be using `/health` or actually `/api/v1/version` for determining readiness, as the server pod isn't really ready to be used until `/api/v1/*` is available, and shouldn't `/health` be used for the liveness check?  And then of course if that's the case then we might need a longer time for readiness, assuming that it takes more time for `/api/v1/*` to deploy then the `/health`.

**Comment by rhuss** [comment]
@gashcrumb I think it makes totally sense to switch to /api/v1/version for the readiness check.
@heiko-braun can we consider this issue for one of the next sprints ?

**Comment by gashcrumb** [comment]
I'd suggest creating a new issue though, that doesn't sound related to the
thing that caused us to open this issue.
